### PR TITLE
enable test scala formatting check in Travis

### PR DESCRIPTION
**Pull request by drcariel**
closes https://github.com/apache/incubator-openwhisk-cli/issues/318

Will cause Travis to fail if the scala tests are not formatted correctly. Considering adding documentation around go and scala fmt here as well

### CODE REVIEW

**Comment by drcariel** [comment]
@rabbah @dubee let me know if there's anything else you'd like me to add here

**Comment by drcariel** [comment]
Note that in this repo it looks like we actually format the go code if it is not formatted correctly - this is not the case for the client-go repo as I found out the hard way...

**Comment by rabbah** [comment]
@chetanmeh mind reviewing this?

File: .travis.yml [hunk 1]
```diff
@@ -51,6 +51,7 @@ install:
 
 before_script:
   - cd $TRAVIS_BUILD_DIR
+  - ./gradlew checkTestScalafmt
```
Comment (chetanmeh) [inline .travis.yml]: Better to run `checkScalafmtAll` i.e. apply check for all source sets

File: build.gradle [hunk 2]
```diff
@@ -15,10 +15,51 @@
...
+
 plugins {
     id 'com.github.blindpirate.gogradle' version '0.8.1'
 }
 
+subprojects {
+  apply plugin: 'scalafmt'
+  scalafmt.configFilePath = gradle.scalafmt.config
+
+  afterEvaluate {
```
Comment (chetanmeh) [inline build.gradle]: This should not be required. In core repo its purpose is to customize the launch classpath for controller and invoker apps. For cli builds we do not require that as there is no module with `application` plugin

File: .travis.yml (hunk 1 above)
Comment (chetanmeh) [reply .travis.yml]: Also use `--console=plain` option

**Pull request merged**